jsondb.js

```diff
@@ -199,6 +199,39 @@ const applyGetMethod = U.when(
 
 export const handleJson = U.compose(buildResponse, applyMethod, applyGetMethod);
 
+const retrievePath = (lensPath, json) => {
+  let restfulPath = restfulLensPath(lensPath, json);
+
+  return !U.isEmpty(restfulPath)
+    ? restfulPath
+    : indexLensPath(lensPath, json);
+};
+
+const indexLensPath = (lensPath, json) => {
+  let finalPath = [];
+
+  for (
+    let pathItem = null, pathCopy = [...lensPath], current = { ...json };
+    (pathItem = pathCopy.shift());
+  ) {
+    if (U.isObject(current)) {
+      current = current[pathItem];
+      finalPath.push(pathItem);
+    } else if (U.isArray(current)) {
+      const index = Number(pathItem);
+      const maybeItem = current[Number(pathItem)];
+      if (isNaN(index) || maybeItem == null) {
+        return [];
+      }
+      current = current;
```

Comment:
Why this?

Reply:
A mistake, it should be `current = maybeItem`.